### PR TITLE
Feature/webpack config tailwindcss

### PR DESCRIPTION
**Pull request by notjb**
Added
---

- [x] Add `@studiometa/webpack-config` to the template
- [x] Add tailwindcss to the template & the corresponding scss
- [x] Add the `postcss.config.js`
- [x] Add a base `tailwind.config.js`

Deleted
---

- [x] Remove the gulp logic
- [x] Remove the Husky pre-commit-hook 

🚀

### CODE REVIEW

**Comment by Guillaume-meta** [comment]
Hello,

Je vois qu’il y a yoast, on testerait pas Rank math comme Ludo l’indiquait ?

A+

File: template/readme.md [hunk 1]
```diff
@@ -8,12 +8,20 @@
...
-### Gulp
+Let's get you started by intilizing the project with `$ npm install`
+Once packages have been installed, start your dev environment with `$ npm run dev` 🚀
+
+### Webpack
 
 ### SCSS
 
+Styles are managed with postCSS and tailwind css.
+The `.scss` files are stored in `/web/wp-content/themes/<%= slug %>/css`.
```
Comment (perruche) [inline template/readme.md]: Path is missing the `src` folder

File: template/tailwind.config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  // Extends the default Studio Meta Tailwind configuration here...
+  plugins: [],
+  theme: {},
+  purge: {
+    // Learn more on https://tailwindcss.com/docs/controlling-file-size/#removing-unused-css
+    enabled: process.env.NODE_ENV === 'production',
+    content: [
+      './web/wp-content/themes/<%= name %>/src/js/**/*.js',
+      './web/wp-content/themes/<%= name %>/src/js/**/*.vue',
+      './web/wp-content/themes/<%= name %>/**/*.tpl',
```
Comment (perruche) [inline template/tailwind.config.js]: I think we can remove `*.tpl` from this list, but maybe we should add `*.php` files ?

File: template/web/wp-content/themes/<%= slug %>/src/css/_config.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*============================================================================*\
+  Config SCSS
+\*============================================================================*/
+
+/* Assets directories
+\*============================================================================*/
+
+$img-dir: '/static/img/';
+$svg-dir: '/static/svg/';
```
Comment (perruche) [inline template/web/wp-content/themes/<%= slug %>/src/css/_config.scss]: `static` folder is in the theme directory right ? Does this work ? shouldn't it be `wp-content/theme/<%= slug %>/static/svg`

File: template/tailwind.config.js (hunk 2 above)
Comment (perruche) [reply template/tailwind.config.js]: I think it should be `<%= slug %>` also in this path no ?

**Review by perruche — CHANGES_REQUESTED** [review]
Good work !

File: template/tailwind.config.js (hunk 2 above)
Comment (notjb) [reply template/tailwind.config.js]: Not sure if adding php files is useful since classes should be used in the view (twig) ?

**Comment by notjb** [comment]
@Guillaume-meta as of today the team as not stated about using YOAST or Math Rank.
It should be added to a dedicated issue.

File: template/tailwind.config.js (hunk 2 above)
Comment (perruche) [reply template/tailwind.config.js]: yes, probably better to leave it like this, and add php files if a project need it

**Comment by perruche** [comment]
>  It should be added to a dedicated issue.

Should we discuss this on the next dev meeting ? @titouanmathis

File: template/package.json [hunk 4]
```diff
@@ -1,40 +1,39 @@
...
     "@studiometa/eslint-config": "^2.0.3",
-    "@studiometa/gulp-config": "^1.5.0-beta.25",
     "@studiometa/prettier-config": "^2.0.0",
-    "@studiometa/stylelint-config": "^1.2.3",
-    "@studiometa/webpack-config": "^1.1.2",
-    "babel-eslint": "^10.0.3",
-    "dotenv": "^8.2.0",
+    "@studiometa/stylelint-config": "^2.0.0",
+    "@studiometa/tailwind-config": "^1.0.0-alpha.9",
+    "@studiometa/webpack-config": "^2.0.0-alpha.6",
```
Comment (titouanmathis) [inline template/package.json]: You can update the `@studiometa/webpack-config` to its latest version.

File: template/postcss.config.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+const postcssConfig = {
+  plugins: [
+    require('postcss-import'),
+    require('tailwindcss'),
+    require('postcss-nested'),
+    require('postcss-custom-properties'),
```
Comment (titouanmathis) [inline template/postcss.config.js:4]: We use SCSS files which are compile with the `sass-loader`, so the `postcss-import` and `postcss-nested` PostCSS plugins are not required.

File: template/postcss.config.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    require('autoprefixer'),
+  ],
+};
+
+if (process.env.NODE_ENV === 'production') {
+  postcssConfig.plugins = [
+    ...postcssConfig.plugins,
+    require('@fullhuman/postcss-purgecss')({
```
Comment (titouanmathis) [inline template/postcss.config.js]: The latest version of Tailwind CSS comes with Purge CSS, we can remove from the PostCSS configuration.

File: template/tailwind.config.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ * @docs    https://tailwindcss.com/docs/configuration
+ * @default https://github.com/studiometa/tailwind-config/blob/develop/src/index.js
+ */
+module.exports = merge(config, {
+  // Extends the default Studio Meta Tailwind configuration here...
+  plugins: [],
+  theme: {},
+  purge: {
+    // Learn more on https://tailwindcss.com/docs/controlling-file-size/#removing-unused-css
+    enabled: process.env.NODE_ENV === 'production',
```
Comment (titouanmathis) [inline template/tailwind.config.js]: This is the default value, we do not need to specify it.

File: template/tailwind.config.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+const merge = require('lodash.merge');
+const plugin = require('tailwindcss/plugin');
```
Comment (titouanmathis) [inline template/tailwind.config.js]: `plugin` is not used, it can be removed.

**Review by titouanmathis — CHANGES_REQUESTED** [review]
Some minor fixes need to be done, but well done!  👍

File: template/tailwind.config.js (hunk 7 above)
Comment (notjb) [reply template/tailwind.config.js]: I'm not sure to understand your feedback, did you mean that also for line 17 ?

Comment (titouanmathis) [reply template/tailwind.config.js]: From the doc:

> Now whenever you compile your CSS with NODE_ENV set to production, Tailwind will automatically purge unused styles from your CSS.

The default behaviour of Tailwind CSS with Purge CSS is to enable it only when `NODE_ENV` is set to `production`. So the configuration of `enabled` (line 17) is redundant with the default one and therefore does not need to be specified.

**Comment by titouanmathis** [comment]
> > It should be added to a dedicated issue.
> 
> Should we discuss this on the next dev meeting ? @titouanmathis

I think someone should test it before, but I will keep this in mind 😉

**Review by titouanmathis — APPROVED** [review]
👍

**Pull request merged**